app/controllers/api/action_plans_controller.rb

```diff
@@ -11,7 +13,9 @@ class Api::ActionPlansController < Api::ApplicationController
   def index
     set_seed
 
-    action_plans = ActionPlan.includes(:revisions, :action_plan_statistics)
+    action_plans = ActionPlan
+      .where(participatory_process_id: @participatory_process.try(:id))
```

Comment:
This is dangerous. There shouldn't be any action plan without participatory process. What about `where(participatory_process: @participatory_process)`.

Also I think we should halt on the `before_filter` if no participatory process was found.